superset-frontend/spec/javascripts/dashboard/components/gridComponents/Tabs_spec.jsx

```diff
@@ -134,7 +134,7 @@ describe('Tabs', () => {
     const wrapper = setup({ editMode: true, onChangeTab });
     wrapper
       .find(
-        '[data-test="dashboard-component-tabs"] .ant-tabs-tab [data-test="short-link-button"]',
+        '[data-test="dashboard-component-tabs"] .ant-tabs-tab [role="button"]',
```

Comment:
Curious the impetus for this lil' change. In this instance, it's a relatively minor example of a pattern question. I  like the data-attributes as it helps us quickly find the (potentially) offending DOM node when a test fails.

Reply:
This change is inspired by React Testing Library principles of accessibility and [query priority](https://testing-library.com/docs/queries/about/#priority). From an accessibility standpoint, the user cannot see or hear test ids so that's why I changed to `role="button"`.

Reply:
Fair enough. I like the data-test pattern mainly because if we see an error in CI, it makes it trivial to find the relevant lines of code (which has been handy with Cypress in particular). This case has a data-test attribute in it anyway, so I'm not worried here.